lib/endpoints/class-wp-rest-terms-controller.php

```diff
@@ -528,6 +489,52 @@ public function delete_item( $request ) {
 	}
 
 	/**
+	 * Prepare a single term for create or update
+	 *
+	 * @param WP_REST_Request $request Request object.
+	 * @return object $prepared_term Term object.
+	 */
+	public function prepare_item_for_database( $request ) {
+		$prepared_term = new stdClass;
+		$prepared_term->args = array();
+
+		if ( isset( $request['name'] ) ) {
+			$prepared_term->name = $request['name'];
+			$prepared_term->args['name'] = $request['name'];
```

Comment:
> but if I return a single term object then would that still work with the wp_insert_term function? I

No, you'll need to break them apart.

> Do you think a better way would be to return a WP_Term object and then supply the entire object as the args and then just use the name attribute for the first parameter of wp_insert_term?

Yes, which is why I left the comment.